core/src/main/java/org/opensearch/sql/expression/operator/arthmetic/ArithmeticFunction.java

```diff
@@ -43,14 +44,27 @@ public class ArithmeticFunction {
    */
   public static void register(BuiltinFunctionRepository repository) {
     repository.register(add());
+    repository.register(addFunction());
     repository.register(subtract());
+    repository.register(subtractFunction());
     repository.register(multiply());
+    repository.register(multiplyFunction());
     repository.register(divide());
+    repository.register(divideFunction());
     repository.register(modules());
+    repository.register(modulus());
+    repository.register(mod());
   }
 
-  private static DefaultFunctionResolver add() {
-    return FunctionDSL.define(BuiltinFunctionName.ADD.getName(),
+  /**
+   * Definition of add(x, y) function.
+   * Returns the number x plus number y
+   * The supported signature of add function is
+   * (x: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE, y: BYTE/SHORT/INTEGER/LONG/FLOAT/DOUBLE)
+   * -> wider type between types of x and y
+   */
+  private static DefaultFunctionResolver addBase(FunctionName functionName) {
+    return FunctionDSL.define(functionName,
```

Comment:
A note for all functions: return type should be wider than arguments.
`2147483647 = INT_MAX`
```
opensearchsql> select 2147483647 + 2147483647;
TransportError(503, 'ArithmeticException', {'error': {'reason': 'There was internal problem at backend', 'details': 'integer overflow', 'type': 'ArithmeticException'}, 'status': 503})
```

Add tests for this for all types.

Reply:
So should functions always return a data type bigger than the input or only if it would overflow? Always returning a bigger input would be easy while only returning a bigger input if needed would take much more work.

Reply:
I think it should always return bigger type. e.g. SHORT + SHORT = INT, regardless of the actual values